src/components/views/Waiter/Waiter.js

```diff
@@ -7,97 +8,125 @@ import TableHead from '@material-ui/core/TableHead';
 
-const demoContent = [
-  {id: '1', status: 'free', order: null},
-  {id: '2', status: 'thinking', order: null},
-  {id: '3', status: 'ordered', order: 123},
-  {id: '4', status: 'prepared', order: 234},
-  {id: '5', status: 'delivered', order: 345},
-  {id: '6', status: 'paid', order: 456},
-];
+class Waiter extends React.Component {
+  static propTypes = {
+    tables: PropTypes.any,
+    fetchTables: PropTypes.func,
+    fetchTableStatusUpdate: PropTypes.func,
+    loading: PropTypes.shape({
+      active: PropTypes.bool,
+      error: PropTypes.oneOfType([PropTypes.bool, PropTypes.string]),
+    }),
+  }
 
-const renderActions = status => {
-  switch (status) {
-    case 'free':
-      return (
-        <>
-          <Button>thinking</Button>
-          <Button>new order</Button>
-        </>
-      );
-    case 'thinking':
-      return (
-        <Button>new order</Button>
-      );
-    case 'ordered':
-      return (
-        <Button>prepared</Button>
-      );
-    case 'prepared':
+  componentDidMount() {
+    const { fetchTables } = this.props;
+    fetchTables();
+  }
+
+  renderActions(id, status) {
+    const { fetchTableStatusUpdate } = this.props;
+
+    switch (status) {
+      case 'free':
+        return (
+          <>
+            <Button onClick={() => fetchTableStatusUpdate(id, 'thinking')}>thinking</Button>
+            <Button onClick={() => fetchTableStatusUpdate(id, 'ordered')} to={`${process.env.PUBLIC_URL}/waiter/order/`}>new order</Button>
+          </>
+        );
+      case 'thinking':
+        return (
+          <>
+            <Button onClick={() => fetchTableStatusUpdate(id, 'ordered')} to={`${process.env.PUBLIC_URL}/waiter/order/new`}>new order</Button>
+          </>
+        );
+      case 'ordered':
+        return (
+          <>
+            <Button onClick={() => fetchTableStatusUpdate(id, 'prepared')}>prepared</Button>
+            <Button to={`${process.env.PUBLIC_URL}/waiter/order/:id`}>new order</Button>
+          </>
+        );
+      case 'prepared':
+        return (
+          <>
+            <Button onClick={() => fetchTableStatusUpdate(id, 'delivered')} to={`${process.env.PUBLIC_URL}/waiter/order/new`}>delivered</Button>
+          </>
+        );
+      case 'delivered':
+        return (
+          <Button onClick={() => fetchTableStatusUpdate(id, 'paid')} to={`${process.env.PUBLIC_URL}/waiter/order/new`}>paid</Button>
+        );
+      case 'paid':
+        return (
+          <Button onClick={() => fetchTableStatusUpdate(id, 'free')} to={`${process.env.PUBLIC_URL}/waiter/order/new`}>free</Button>
+        );
+      default:
+        return null;
+    }
+  }
+
+  render() {
+    const { loading: { active, error }, tables } = this.props;
+
+    if (active || !tables.length) {
       return (
-        <Button>delivered</Button>
+        <Paper className={styles.component}>
+          <p>Loading...</p>
+        </Paper>
       );
-    case 'delivered':
+    } else if (error) {
```

Comment:
Jak wewnątrz ifa robisz return, to możesz pominąć `else if` oraz ostatniego `elsa`, bo return zawsze kończy wykonanie metody, a kod dzięki temu stanie się czytelniejszy